shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -56,28 +89,82 @@ function _getVerifiedStorageObject(storageType: StorageType): Storage {
  */
 function _getSessionStorageObject(): Storage {
     if (utlCanUseSessionStorage()) {
-        return _getVerifiedStorageObject(StorageType.SessionStorage);
+        return _verifiedSessionStorage.value(() => _getVerifiedStorageObject(StorageType.SessionStorage));
     }
 
     return null;
 }
 
+/**
+ * Helper function to safely wrap storage operations in try/catch
+ * and reset cache if an operation fails
+ * @param storageFn - Function that performs a storage operation
+ * @param storageType - Type of storage being accessed
+ * @param fallbackValue - Value to return if operation fails
+ * @param logger - Optional logger for error reporting
+ * @returns Result of the storage operation or the fallback value
+ */
+function _safeStorageOperation<T>(storageFn: () => T, storageType: StorageType, fallbackValue: T, logger?: IDiagnosticLogger, errorMessageId?: _eInternalMessageId, errorMessage?: string): T {
+    try {
+        return storageFn();
+    } catch (e) {
+        // If operation fails, invalidate the cache
+        if (storageType === StorageType.LocalStorage) {
+            _canUseLocalStorage = false;
+            _verifiedLocalStorage.reset();
+            
+            // Log error if logger is provided
+            if (logger && errorMessageId) {
+                _throwInternal(
+                    logger,
+                    eLoggingSeverity.WARNING,
+                    errorMessageId,
+                    errorMessage + " " + getExceptionName(e),
+                    { exception: dumpObj(e) }
+                );
+            }
+        } else {
+            _canUseSessionStorage = false;
+            _verifiedSessionStorage.reset();
+            
+            // Log error if logger is provided
+            if (logger && errorMessageId) {
+                _throwInternal(
+                    logger,
+                    eLoggingSeverity.WARNING,
+                    errorMessageId,
+                    errorMessage + " " + getExceptionName(e),
+                    { exception: dumpObj(e) }
+                );
+            }
+        }
+        return fallbackValue;
+    }
+}
+
 /**
  * Disables the global SDK usage of local or session storage if available
  */
 export function utlDisableStorage() {
     _canUseLocalStorage = false;
     _canUseSessionStorage = false;
+    _verifiedLocalStorage.reset();
```

Comment:
There is no "reset", just set to null

Reply:
And use the face that this is set to null to determine that a check needs to be performed, then the "_canUseXXXXStorage" can simply use !!_verifiedXXX.v to determine if it can access and for the explicit case of utlDisableStorage it cause use _verifiedXXXXStorage = createCachedValue(null)